lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -136,6 +128,54 @@ def build_client(options)
       Elasticsearch::Client.new(client_options)
     end
 
+    HOSTNAME_PORT_REGEX=/\A(?<hostname>([A-Za-z0-9\.\-]+))(:(?<port>\d+))?\Z/
```

Comment:
What do you think of this regex?
`\A(?<hostname>([A-Za-z0-9\.\-]+)|\[[0-9\:]+\])(:(?<port>\d+))?\Z`